### DISCUSSION TITLE
[@types/react-router-dom] TS2305: React Router has no exported member

### DISCUSSION THREAD

**Discussion by TR909DRUM** [DefinitelyTyped/DefinitelyTyped#62593] [Issues with a @types package]
### What version of React Router are you using?

6.4.1 - I also **tested with version 5.3.4, and 5.2.0 and it keeps getting errors that look like the one from below. (TS2305)**

### Steps to Reproduce

I am getting the following error and don't know the cause of this problem. 

    TS2305: Module '"../../../../../../../../../Users/users1039/IdeaProjects/sampleProject/client/node_modules/@remix-run/router/dist/utils"' has no exported member 'ParamParseKey'.

Here is my tsconfig: 

        {
      "compilerOptions": {
        "outDir": "build/dist",
        "module": "esnext",
        "target": "es5",
        "lib": ["es6", "dom"],
        "sourceMap": true,
        "allowJs": true,
        "jsx": "react",
        "moduleResolution": "node",
        "rootDir": "src",
        "forceConsistentCasingInFileNames": true,
        "noImplicitReturns": true,
        "noImplicitThis": true,
        "noImplicitAny": true,
        "strictNullChecks": false,
        "suppressImplicitAnyIndexErrors": true,
        "noUnusedLocals": true,
        "baseUrl": "."
      },
      "exclude": [
        "node_modules",
        "build",
        "scripts",
        "acceptance-tests",
        "webpack",
        "jest",
        "src/setupTests.ts"
      ]

}

package.json

    {
      "name": "client",
      "version": "1.1.0",
      "private": true,
      "dependencies": {
        "@types/jest": "25.2.1",
        "@types/node": "13.13.2",
        "@types/prop-types": "15.7.3",
        "@types/react": "16.9.34",
        "@types/react-dom": "16.9.6",
        "@types/react-numeric-input": "2.2.3",
        "dotenv": "8.2.0",
        "loadbalance": "1.0.0",
        "moment": "2.24.0",
        "npm": "6.14.4",
        "react": "16.13.1",
        "react-bootstrap": "1.0.0",
        "react-datetime": "2.16.3",
        "react-dom": "16.13.1",
        "react-router-dom": "^6.4.1",
        "react-numeric-input": "2.2.3",
        "react-scripts-ts": "4.0.8",
        "react-scroll-up-button": "1.6.4",
        "typescript": "3.8.3"
      },
      "scripts": {
        "start": "react-scripts-ts start",
        "build": "react-scripts-ts build",
        "test": "react-scripts-ts test --env=jsdom",
        "eject": "react-scripts-ts eject"
      },
      "browserslist": [
        ">0.2%",
        "not dead",
        "not ie <= 11",
        "not op_mini all"
      ]
    }

### Expected Behavior

Trying to get the correct behavior as seen in the examples from: https://reactrouter.com/docs/en/v6/examples/basic

### Actual Behavior

TS2305: Module '"../../../../../../../../../Users/users1039/IdeaProjects/sampleProject/client/node_modules/@remix-run/router/dist/utils"' has no exported member 'ParamParseKey'.

[a comment by an automated account is not shown]

**Comment by eps1lon** [comment]
This looks like a configuration issue in your `tsconfig.json` or a bad dependency tree (e.g. `@types/react-router-dom` being added more than once).

Without having the full app, it's next to impossible to understand what went wrong. But rest assured that error message is misleading. I would suggest asking in the [TypeScript community Discord](https://discord.gg/9axK9G9n) for support.
